src/lsdb/catalog/catalog.py

```diff
@@ -444,3 +449,51 @@ def join(
         )
         hc_catalog = hc.catalog.Catalog(new_catalog_info, alignment.pixel_tree, schema=get_arrow_schema(ddf))
         return Catalog(ddf, ddf_map, hc_catalog)
+
+    def join_nested(
+        self,
+        other: Catalog,
+        left_on: str | None = None,
+        right_on: str | None = None,
+        nested_column_name: str | None = None,
+        output_catalog_name: str | None = None,
+    ) -> Catalog:
+        """Perform a spatial join to another catalog
```

Comment:
Add more to this docstring to explain why it's different from the `join` operation.